fix(logging): improve python core's debug logging

padamstx opened the pull request.

pyrooka commented on ibm_cloud_sdk_core/authenticators/bearer_token_authenticator.py:
```diff
@@ -15,9 +15,11 @@
 # limitations under the License.
 
 from requests import Request
-
```
This empty line should be there to separate 3rd party and local package imports. :)

padamstx replied:
fixed

pyrooka commented on ibm_cloud_sdk_core/authenticators/iam_request_based_authenticator.py:
```diff
@@ -15,11 +15,13 @@
 # limitations under the License.
 
 from typing import Dict
-
```
This empty line should be there to separate std lib and 3rd party package imports. :)

padamstx replied:
fixed

pyrooka reviewed (commented):
I think there should be 2 empty lines between the imports and the `logger` declaration. IIRC top level code blocks should be surrounded by 2 empty lines. Feel free to correct me if I'm wrong!

pyrooka reviewed (approved):
LGTM!

dpopp07 commented on test/utils/logger_utils.py:
```diff
@@ -0,0 +1,35 @@
+    """Sets up logging with a console handler to assist test cases."""
+    logger = get_logger()
+    logger.setLevel(level)
+    # if not logger.hasHandlers():
```
Is this comment outdated?

padamstx replied:
probably :)   I'll fix it before merging.

dpopp07 reviewed (approved):
Looks good! 👍

pyrooka commented:
As I'm looking at the changes and the README. Shouldn't we update the logging section of the README? Especially the code below. Now it's enough to enable the debug logging before the base service is created, right?
```markdown
### Low level request and response dump
To get low level information of the requests/ responses:

from http.client import HTTPConnection
HTTPConnection.debuglevel = 1
```

padamstx commented:
> As I'm looking at the changes and the README. Shouldn't we update the logging section of the README? Especially the code below. Now it's enough to enable the debug logging before the base service is created, right?
> 
> ```
> ### Low level request and response dump
> To get low level information of the requests/ responses:
> 
> from http.client import HTTPConnection
> HTTPConnection.debuglevel = 1
> ```

Yes, I'm planning on doing that prior to merging.

padamstx commented:
@dpopp07 @pyrooka I think I can say that this PR is finally finished! :joy:  In my more recent changes, I've simplified a bit how testcases can enable debug logging, and I also updated the README.md to have more up-to-date info about logging.

pyrooka reviewed (approved):
I think I went through all the changes that you made since my last review, and I still think this looks good! :)

dpopp07 reviewed (approved):
New changes look good! 👍

dpopp07 commented:
Looks pretty similar so maybe it's good to go already but does this have any implications for how the _SDKs_ do logging, as documented [here](https://github.com/IBM/ibm-cloud-sdk-common?tab=readme-ov-file#logging)?

padamstx commented:
> Looks pretty similar so maybe it's good to go already but does this have any implications for how the _SDKs_ do logging, as documented [here](https://github.com/IBM/ibm-cloud-sdk-common?tab=readme-ov-file#logging)?

Yes, I have a to-do item to go through the logging-related info in the common sdk repo and make adjustments as needed after all the logging-related changes in the cores.

ibm-devx-sdk commented:
:tada: This PR is included in version 3.21.0 :tada:

The release is available on [GitHub release](https://github.com/IBM/python-sdk-core/releases/tag/v3.21.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
